tensorflow_probability/python/distributions/pert.py

```diff
@@ -0,0 +1,164 @@
+  parameter, smoothness (g).
+
+  Smoothness determines the properties of Beta distribution:
+
+  ```
+  concentration1 = 1 + g * (MODE - MIN)/(MAX - MIN)
+  concentration0 = 1 + g * (MAX - MODE)/(MAX - MIN)
+  mean = (MIN + g * MODE + MAX)/(g + 2)
+  ```
+
+  PERT distribution is obtained when g = 4.
+
+  For general reference: https://en.wikipedia.org/wiki/PERT_distribution
+  """
+  def __init__(self,
+               mini,
+               mode,
+               maxi,
+               smoothness=4.,
+               validate_args=False,
+               allow_nan_stats=False,
+               name="PERT"):
+    parameters = dict(locals())
+    with tf.name_scope(name) as name:
+      dtype = dtype_util.common_dtype([mini, maxi, mode], tf.float32)
+      self._min = tensor_util.convert_nonref_to_tensor(
+          mini, name='mini', dtype=dtype)
+      self._mod = tensor_util.convert_nonref_to_tensor(
+          mode, name='mode', dtype=dtype)
+      self._max = tensor_util.convert_nonref_to_tensor(
+          maxi, name='maxi', dtype=dtype)
+      self._smoothness = tensor_util.convert_nonref_to_tensor(
+          smoothness, name='smoothness', dtype=dtype)
+      self._concentration1 = 1. + self._smoothness \
+        * (self._mod - self._min) / (self._max - self._min)
+      self._concentration0 = 1. + self._smoothness \
+        * (self._max - self._mod) / (self._max - self._min)
+
+      self._scale = self._max - self._min
+
+      # Broadcasting scale on affine bijector to match batch dimension.
+      # This prevents dimension mismatch for operations like cdf.
+      super(PERT, self).__init__(
+          distribution=Beta(
+              concentration1=self._concentration1,
+              concentration0=self._concentration0,
+              allow_nan_stats=allow_nan_stats),
+          bijector=AffineScalar(
+              shift=self._min,
+              scale=tf.broadcast_to(
+                  input=self._scale,
+                  shape=self._batch_shape_tensor(
+                      concentration1=self._concentration1,
+                      concentration0=self._concentration0))),
+          validate_args=validate_args,
+          parameters=parameters,
+          name=name)
+
+  def _batch_shape_tensor(self, concentration1=None, concentration0=None):
+    return prefer_static.broadcast_shape(
+        prefer_static.shape(
+            self.concentration1 if concentration1 is None else concentration1),
+        prefer_static.shape(
+            self.concentration0 if concentration0 is None else concentration0))
+
+  def _parameter_control_dependencies(self, is_init):
+    if not self.validate_args:
+      return []
+    assertions = []
+    if is_init != tensor_util.is_ref(self._mod):
+      if is_init != tensor_util.is_ref(self._min):
+        assertions.append(assert_util.assert_greater(
+            self._mod,
+            self._min,
+            message="Mode must be greater than minimum."
```

Comment:
minor nit: can we put these param names in backticks like "`peak` must be greater than `low`"